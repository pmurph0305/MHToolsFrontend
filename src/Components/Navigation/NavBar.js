import logo from './mh_logo_white.png';
import NavButtonLi from '../NavButton/NavButtonLi';
import React from 'react';

const NavBar = (props) => {
    return(
        <div>
            <header className='bg-black w-100 pv0-ns'>
                <nav className="f6 fw6 ttu tracked"
                    role='navigation'
                    aria-labelledby='navlabel'
                    >
                    <h2 id='navlabel'>Navigation links</h2>
                    <img 
                        className="v-mid dib h2 w2 pointer" 
                        alt='logo' 
                        src={logo}
                        onClick={() => props.onRouteChange('home')}
                    />
                    <NavButtonLi
                        buttonLabel="Daily Maintenance"
                        onRouteChange={props.onRouteChange}
                        route='dm'
                        
                    />
                    <NavButtonLi
                        buttonLabel="PHQ-9"
                        onRouteChange={props.onRouteChange}
                        route='phq9'
                        
                    />
                    <NavButtonLi
                        buttonLabel="CBT"
                        onRouteChange={props.onRouteChange}
                        route='cbt'
                    />
                    <NavButtonLi
                        buttonLabel="Coping Skills"
                        onRouteChange={props.onRouteChange}
                        route='coping'
                    />
                    <NavButtonLi
                        buttonLabel="History"
                        onRouteChange={props.onRouteChange}
                        route='hist'
                    />
                    <NavButtonLi
                        buttonLabel="Sign in"
                        liClass='fr'
                        onRouteChange={props.onRouteChange}
                        route='signin'
                    />
                </nav>
            </header>
        </div>
    )
}

export default NavBar;