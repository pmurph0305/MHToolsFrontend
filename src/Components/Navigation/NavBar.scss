@import "../../styles/shared.scss";

.navHeader {
	background-color: $button-background-color;
    width: 100%;
    // font sizes done for all nav elements based on screen sizes.
    @media screen and (min-width: 60em) {
        font-size:1rem;
    }
    // Medium
    @media screen and (min-width: 30em) and (max-width: 60em) {	
        font-size:0.75rem;
    }
    // Small
    @media screen and (min-width: 20em) and (max-width: 30em) {	
        font-size:.55rem;
	}


	// Nav elements in an ul
	ul {
		// each one is an li.
		li {
			cursor: pointer;
			display: inline-block;
			// far left nav is an image to click to go home.
			img {
                padding-left: 0.5rem;
				vertical-align: middle;
				// Large
				@media screen and (min-width: 60em) {
                    height: 2rem;
                    width: 2rem;
				}
				// Medium
				@media screen and (min-width: 30em) and (max-width: 60em) {
					height: 1.5rem;
                    width: 1.5rem;
				}
				// Small
				@media screen and (max-width: 30em) {
					height: 1rem;
                    width: 1rem;
                }
				&:focus,
				&:hover,
				&:active {
					cursor: pointer;
					opacity: 0.5;
					transition: opacity 0.15s ease-in;
				}
			}
            // the rest of them are buttons in an li
			// slightly increased opacity in transition vs normal buttons.
			button {
				@extend %ButtonBase;
				&:focus,
				&:hover,
				&:active {
					opacity: 0.7;
                }
                // Large
				@media screen and (min-width: 60em) {
                    padding: 1rem;
				}
				// Medium
				@media screen and (min-width: 30em) and (max-width: 60em) {
					padding: 0.65rem;
				}
				// Small
				@media screen and (min-width: 20em) and (max-width: 30em) {
                    padding: 0.4rem;
				}
			}
		}
		// last li element is signin/logout so float right.
		// also added register to float right
		li:last-child, .register {
			float: right;
		}
		
		

	}
}
