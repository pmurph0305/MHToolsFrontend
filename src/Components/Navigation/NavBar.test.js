import { shallow } from "enzyme";
import React from "react";
import NavBar from "./NavBar";

describe("Navbar tests", () => {
  let mockRouteChange = jest.fn(val => val);

  const wrapper = shallow(<NavBar onRouteChange={mockRouteChange} />);

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("route change is called", () => {
    wrapper.find("img").simulate("click");
    expect(mockRouteChange).toHaveBeenCalledTimes(1);
    expect(mockRouteChange.mock.results[0].value).toBe("home");
  });

  it("displays correct number of nav buttons", () => {
    expect(wrapper.find("NavButtonLi").length).toBe(7);
  });
});
