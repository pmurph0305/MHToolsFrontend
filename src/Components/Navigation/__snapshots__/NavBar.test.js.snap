// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests matches snapshot 1`] = `ShallowWrapper {}`;
